### PR TITLE
Introduced non-generic type descriptor method `descriptor.Type(System.Type)`.

### LINKED ISSUE #1079: Introduce non-generic type descriptor method `descriptor.Type(System.Type).`
#1058

For dynamic schema creating it would be useful to have a new `Type` description method that can be used to dynamically set types.

We need to write some validation logic that ensures the type context is always correct (input vs output).

Places where we have a type context:

- Object Fields
- Object Field Arguments
- Interface Fields
- Interface Field Arguments
- Input Fields
- Directive Arguments

### PR DESCRIPTION
**Pull request by promontis**
Addresses #1079

### CODE REVIEW

File: src/Core/Types/Types/Descriptors/ArgumentDescriptorBase~1.cs [hunk 1]
```diff
@@ -45,6 +45,20 @@ public void Type<TInputType>()
                 TypeContext.Input);
         }
 
+        public void Type(Type type)
```
Comment (michaelstaib) [inline src/Core/Types/Types/Descriptors/ArgumentDescriptorBase~1.cs]: The above method should now lean on this non-generic method.

File: src/Core/Types/Types/Descriptors/OutputFieldDescriptorBase.cs [hunk 2]
```diff
@@ -55,6 +55,20 @@ protected void Type<TOutputType>()
                 TypeContext.Output);
         }
 
+        protected void Type(Type type)
+        {
```
Comment (michaelstaib) [inline src/Core/Types/Types/Descriptors/OutputFieldDescriptorBase.cs]: The above method should now lean on this new non-generic method.

**Comment by promontis** [comment]
@michaelstaib thanks for having a quick look! I'll add the tests, pull to the interface and resolve you comments! Probably tonight or tomorrow

File: src/Core/Types/Types/Descriptors/ArgumentDescriptorBase~1.cs (hunk 1 above)
Comment (promontis) [reply src/Core/Types/Types/Descriptors/ArgumentDescriptorBase~1.cs]: added

File: src/Core/Types/Types/Descriptors/OutputFieldDescriptorBase.cs (hunk 2 above)
Comment (promontis) [reply src/Core/Types/Types/Descriptors/OutputFieldDescriptorBase.cs]: added

**Comment by promontis** [comment]
@michaelstaib I've added tests, pulled to interface, resolved the comments and fixed a bug.

Is it ok like this?

Btw, I didn't see any `OutputFieldDescriptorTests.cs`. Do you want me to add it?

**Comment by promontis** [comment]
@michaelstaib is it ready to be merged :) ? If so, could you also build a preview nuget?

**Comment by michaelstaib** [comment]
@promontis sorry was in a meeting ... yes everything is green :) I will trigger the merge and release.

**Pull request merged**